### PR TITLE
DDP-7109 add Edit user preferences dialog into basil project and add birthday&username fields

### PR DESCRIPTION
**Pull request by aglushko-broad**
image

### CODE REVIEW

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 1]
```diff
@@ -182,6 +182,14 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
      */
     @Output()
     public valueChanged = new EventEmitter<Address | null>();
+
+    /**
+     * Will emit an address whenever it is updated from ddp-address-input component
+     * type {EventEmitter<Address>}
+     */
+    @Output()
+    public inputAddress = new EventEmitter<Address | null>();
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: The address that comes from `ddp-address-input` component is meant to remain private within the embedded address. Let's not externalize this. It is not a clean address/not meant to be seen outside of addressEmbedded.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/user/userPreferences.component.ts [hunk 2]
```diff
@@ -170,62 +153,62 @@ export class UserPreferencesComponent implements OnDestroy {
...
+        return this.loaded && this.isAddressValid
+            // let's allow to save preferences if user didn't touch empty birthday date since s/he can fill it later at action forms
+            && (this.birthDate.valid || (this.birthDate.invalid && this.birthDate.pristine));
     }
 
     public save(): void {
-        if (!this.isDateReadyForSave(true)) {
-            this.showError = true;
+        if (!(this.profileForm.dirty || this.addressDirty)) {
+            this.dialogRef.close();
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/user/userPreferences.component.ts]: Would checking that address component is busy sufficient to decide whether we can close the dialog?

Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/user/userPreferences.component.ts]: you're right, we don't need this block, removed it

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts (hunk 1 above)
Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: done

File: ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts [hunk 3]
```diff
@@ -182,6 +182,14 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
      */
     @Output()
     public valueChanged = new EventEmitter<Address | null>();
+
+    /**
+     * Will emit an address whenever it is updated from ddp-address-input component
+     * type {EventEmitter<Address>}
+     */
+    @Output()
+    public inputAddress = new EventEmitter<void>();
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: This is still here. Are you sure you  pushed your changes that depend on mailing address being busy?

Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: As far as I understand from your previous comment we want to keep address private and we do it. We just emit the empty event with no data. But I use it in the parent to find out that smth was changed and mark the component dirty

Comment (aglushko-broad) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts]: fixed

**Review by alextanlan — APPROVED** [review]

**Pull request merged**